test/fail_compilation/test21008.d

```diff
@@ -0,0 +1,42 @@
+/*
+DISABLED: freebsd32 linux32 osx32 win32
```

Comment:
Why is this platform-specific?

Reply:
Because the error messages rely on the size of size_t, so restricting it to 64 bits means the error messages match.

Reply:
That's why `TEST_OUTPUT` supports conditional output, e.g. `$?:32=uint|64=ulong$`. [README](https://github.com/dlang/dmd/blob/master/test/README.md#test_output).

Reply:
Ok, given the rationale, please change to what @MoonlightSentinel suggested.